lib/linter-clang.coffee

```diff
@@ -1,213 +1,87 @@
-
-    # use BufferedNodeProcess if the linter is node executable
-    if @isNodeExecutable
-      Process = BufferedNodeProcess
-    else
-      Process = BufferedProcess
-
-    # options for BufferedProcess, same syntax with child_process.spawn
-    options = {cwd: @cwd}
-
-    result = []
-
-    aggregate = (messages) ->
-      result = result.concat messages
-
-    stdout = (output) =>
-      if atom.inDevMode()
-        console.log 'clang: stdout', output
-      if @errorStream == 'stdout'
-        @processMessage(output, aggregate)
-
-    stderr = (output) =>
-      if atom.inDevMode()
-        console.warn 'clang: stderr', output
-      if @errorStream == 'stderr'
-        @processMessage(output, aggregate)
-
-    if atom.inDevMode()
-      console.log "linter-clang: command = #{command}, args = #{args}, options = #{options}"
-
-    exit = () ->
-      callback result
-
-    new Process({command, args, options, stdout, stderr, exit})
-
-  constructor: (editor) ->
-    @editor = editor
-
-    if editor.getGrammar().name == 'C++'
-      @language = 'c++'
-      # @flag = '-std=c++11'
-    if editor.getGrammar().name == 'Objective-C++'
-      @language = 'objective-c++'
-      # @flag = ''
-    if editor.getGrammar().name == 'C'
-      @language = 'c'
-      # @flag = ''
-    if editor.getGrammar().name == 'Objective-C'
-      @language = 'objective-c'
-      # @flag = ''
-
-    super(editor)
-
-  createMessage: (match) ->
-    # message might be empty, we have to supply a value
-    if match and match.type == 'parse' and not match.message
-      message = 'error'
-
-    super(match)
-
-module.exports = LinterClang
+module.exports = LinterClang =
+  config:
+    command:
+      type: 'string'
+      default: 'clang'
+    includePaths:
+      type: 'string'
+      default: '.'
+    suppressWarnings:
+      type: 'boolean'
+      default: false
+    defaultCFlags:
+      type: 'string'
+      default: '-Wall'
+    defaultCppFlags:
+      type: 'string'
+      default: '-Wall -std=c++11'
+    defaultObjCFlags:
+      type: 'string'
+      default: ' '
+    defaultObjCppFlags:
+      type: 'string'
+      default: ' '
+    errorLimit:
+      type: 'integer'
+      default: 0
+    verboseDebug:
+      type: 'boolean'
+      default: false
+
+  activate: ->
+    unless atom.packages.getLoadedPackages 'linter-plus'
+      atom.notifications.addError '[linter-clang] `linter-plus` package not found, please install it'
+
+  provideLinter: -> {
+    grammarScopes: ['source.c', 'source.cpp', 'source.objc', 'source.objcpp']
+    scope: 'file'
+    lint: @lint
+    lintOnFly: false
+  }
+
+  lint: (TextEditor) ->
+    regex = ///
+      (.+): #File with issue
+      (\d+): #Line with issue
+      (\d+):\s #Column with issue
+      (.+):\s+ #Type of issue
+      (.*) #Message explaining issue
+    ///
+    return new Promise (Resolve) ->
+      filePath = TextEditor.getPath()
+      if filePath # Files that have not be saved
```

Comment:
This is redundant, https://github.com/AtomLinter/linter-ruby/pull/18/files#r33106620